src/napari_imagej/_module_utils.py

```diff
@@ -395,20 +398,48 @@ def _modify_function_signature(
         ]
         all_params = module_params + other_params
         function.__signature__ = sig.replace(
-            parameters=all_params, return_annotation=_return_type(module_info)
+            parameters=all_params, return_annotation=List[LayerDataTuple]
         )
     except Exception as e:
         print(e)
 
 
-def _module_output(module: "jc.Module") -> Any:
+def _layerDataTuple_from_layer(layer: Layer):
+    data = layer.data
+    metadata: Dict[str, Any] = {}
+    layer_type = type(layer).__name__
+    metadata["name"] = layer.name
+    metadata["metadata"] = layer.metadata
+    if hasattr(layer, "shape_type"):
+        metadata["shape_type"] = getattr(layer, "shape_type")
+
+    return (data, metadata, layer_type)
+
+
+def _module_outputs(module: "jc.Module") -> Tuple[List[LayerDataTuple], List[Any]]:
     """Gets the output of the module, or None if the module has no output."""
+    layerDataTuples = []
+    otherOutputs = []
```

Comment:
Is there a better name for `otherOutputs`? :laughing:

I dislike `non_layer_outputs`, but I'd think we can do better.

Reply:
Fixed up in 8b35530